src/JuMP.jl

```diff
@@ -269,30 +269,38 @@ if VERSION >= v"0.7-"
 end
 
 
-# In Automatic and Manual mode, `model.moi_backend` is either directly the
+# In Automatic and Manual mode, `backend(model)` is either directly the
 # `CachingOptimizer` if `bridge_constraints=false` was passed in the constructor
 # or it is a `LazyBridgeOptimizer` and the `CachingOptimizer` is stored in the
 # `model` field
 function caching_optimizer(model::Model)
-    if model.moi_backend isa MOIU.CachingOptimizer
-        return model.moi_backend
-    elseif (model.moi_backend isa
+    if backend(model) isa MOIU.CachingOptimizer
+        return backend(model)
+    elseif (backend(model) isa
             MOI.Bridges.LazyBridgeOptimizer{<:MOIU.CachingOptimizer})
-        return model.moi_backend.model
+        return backend(model).model
     else
         error("The function `caching_optimizer` cannot be called on a model " *
               "in `Direct` mode.")
     end
 end
 
+"""
+    backend(model::Model)
+
+Return the MathOptInterface model. This should only be used by advanced users
```

Comment:
Which MathOptInterface model? There are typically lots of them. The docstring to point to the documentation on manual/automatic/direct modes to explain what object you get when you ask for the backend.

Reply:
I made a stab, but honestly, there are so many layers I'm not 100% sure I got it right.